code/game/objects/items/stacks/stack_recipes.dm

```diff
@@ -506,6 +522,93 @@ var/list/datum/stack_recipe/cardboard_recipes = list (
+	new/datum/stack_recipe/cloth("Cleaning Rag",	/obj/item/weapon/reagent_containers/glass/rag,	1,	time = 20),
+	new/datum/stack_recipe/cloth("Toga",			/obj/item/clothing/under/toga,					3,	time = 50),
+	new/datum/stack_recipe/cloth("Bedsheet",		/obj/item/weapon/bedsheet/linen,				2,	time = 20),
+	)
+
+//keep in mind that tool crafting time is reduced by x0.75 with needles and x0.5 with a sewing machine, then all the way down to x0.1 with upgrades
+//a rule of thumb I settled on is 40 ticks per cloth used for the recipe
+var/list/datum/stack_recipe/cloth_recipes_with_tool = list (
+	null,
+	"Uniforms",
+	new/datum/stack_recipe/cloth("Jumpsuit",				/obj/item/clothing/under/color/linen,			5,	time = 200),
+	new/datum/stack_recipe/cloth/composite("Composite Set",	/obj/item/clothing/under/composite,2),
+	new/datum/stack_recipe/cloth("Sleeve-less Dress",		/obj/item/clothing/under/dress,					4,	time = 160),
+	new/datum/stack_recipe/cloth("Villager Dress",			/obj/item/clothing/under/villager_dress,		5,	time = 200),
+	"Suits",
+	new/datum/stack_recipe/cloth("Labcoat",					/obj/item/clothing/suit/storage/labcoat/linen,	3,	time = 120),
+	"Hats",
+	new/datum/stack_recipe/cloth("Soft Cap",				/obj/item/clothing/head/soft/linen,				2,	time = 80),
+	new/datum/stack_recipe/cloth("Flat Cap",				/obj/item/clothing/head/flatcap/linen,			2,	time = 80),
+	new/datum/stack_recipe/cloth("Ushanka",					/obj/item/clothing/head/ushanka/linen,			3,	time = 120),
+	"Masks",
+	new/datum/stack_recipe/cloth("Ski Mask",				/obj/item/clothing/mask/balaclava/skimask/linen,2,	time = 80),
+	new/datum/stack_recipe/cloth("Scarf",					/obj/item/clothing/mask/scarf/linen,			1,	time = 40),
+	"Gloves",
+	new/datum/stack_recipe/cloth("Mittens",					/obj/item/clothing/gloves/mittens,				2,	time = 80),
+	"Accessories",
+	new/datum/stack_recipe/cloth("Tie",						/obj/item/clothing/accessory/tie/linen,			1,	time = 40),
+	new/datum/stack_recipe/cloth("Armband",					/obj/item/clothing/accessory/armband/linen,		1,	time = 40),
+	)
+
+/datum/stack_recipe/cloth/composite/before_build(var/mob/user)
+	//first we pick some pants
+	extra_data = list()
+	time = 0
+	req_amount = 0
+	var/list/available_pants = list(
+		"Short Pants (2 cloth)" = list("shortpants",2),
+		"Long Pants (3 cloth)" = list("pants",3),
+		"Tartan Kilt (2 cloth)" = list("tartankilt",2),
+		"Pleated Skirt (3 cloth)" = list("pleatedskirt",3),
+		"Straight Skirt (2 cloth)" = list("straightskirt",2),
+		)
+	var/choice = input(user, "What kind of pants?","Composite Set",null) as null|anything in available_pants
+	if (!choice)
+		req_amount = 2
+		return FALSE
+
+	var/list/result = available_pants[choice]
+	extra_data += result[1]
+	req_amount += result[2]
+
+	//then we may pick a top or none
+	var/list/available_tops = list(
+		"None" = null,
+		"Polo (2 cloth)" = list("polo",2),
+		"T-Shirt (2 cloth)" = list("tshirt",2),
+		)
+	choice = input(user, "What kind of top?","Composite Set",null) as null|anything in available_tops
```

Comment:
yeah I forgot to have null taken into account for the second choice, it'll do the same as choosing "None", and you'll sew pants without top. If you cancel above it will just cancel the crafting altogether.